client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -242,9 +245,19 @@ export class PlanFeatures2023Grid extends Component<
 	PlanFeatures2023GridType,
 	PlanFeatures2023GridState
 > {
-	state = {
-		showPlansComparisonGrid: false,
-	};
+	constructor( props: PlanFeatures2023GridType ) {
+		super( props );
+
+		this.state = {
+			showPlansComparisonGrid: false,
+			selectedStorage: {},
+		};
+
+		this.props.planProperties.forEach( ( planProperties ) => {
+			const { planName } = planProperties;
+			this.state.selectedStorage[ planName ] = '';
```

Comment:
Did nothing flag the direct mutation of state here? You probably shouldn't be doing this.

Reply:
> Did nothing flag the direct mutation of state here?

Nothing flagged this -- I imagine it was because this is done in the constructor when the component is first initialized. More about why explained in https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1256434255

Open to suggestions though

Reply:
ok. I'm seeing this as something we shouldn't do except directly in constructor (not in a forEach callback). I could be wrong though, but worth investigating. If you do need to initialize these values to something (probably not empty string either, but `null`), then I'd maybe create the object and insert it to state after. something like

```
const foo = ... forEach( ( properties ...
this.state = { foo ...
```

Reply:
[Going to try tinkering more](https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1256535482) before going down this route

Reply:
So I played around a bit more. Initialization with the forEach was not necessary, and I conflated it with another change that fixed the typescript error I was seeing. I remove it in [b64d2db](https://github.com/Automattic/wp-calypso/pull/79041/commits/b64d2db02f4d4ca8f2a788383110965a88785ab1)

What matters is actually initializing state in the constructor, rather than as a class member

https://github.com/Automattic/wp-calypso/pull/79041/files#diff-e268185bf1c246017f6796f5f74ff3c0ae49b12e2acf525d8074859449573a19L245-R255

I'm not quite sure what to make of it, or if moving state initialization to the constructor affects any behavior. 🤔 Some preliminary googling doesn't hint at anything meaningful either